packages/front/svelte/components/SubventionsVersementsDashboard/SubventionTable/SubventionTable.svelte

```diff
@@ -64,7 +64,9 @@
                         {elementData.montantsDemande}
                     </TableCell>
                     <TableCell position="end" primary="true">
-                        <StatusLabel status={elementData.status} />
+                        {#key elementData.status}
```

Comment:
Pourquoi un #key ? 

La valeur peut changer dynamiquement ?

Reply:
"Key blocks destroy and recreate their contents when the value of an expression changes."

Reply:
Ben justement quand on change d'année il garde le composant et c'est cause de pb

Reply:
Mais quand on change d'année c'est tout le tableau qui est re-render non ?

Reply:
> Mais quand on change d'année c'est tout le tableau qui est re-render non ?

Ben justement non, svelte doit optimiser des trucs et on doit le forcer à recalculer `StatusLabel`

Reply:
Ce qui me semble bizzare c'est qu'on doivent le faire que sur cette element

Reply:
Peut etre que c'est status label qui n'est pas recréer ! Et vu qu'on set la var dans le controller et que c'est cette var qu'on uttilise en suite ! La modification du status par svelte n'a pas d'impacte sur l'affichage !

Reply:
Je vois deux solution sois uttiliser en direct la property dans le componant svelte sois avoir un store qui ecoute le changement de valeur ! Je pencherai pour la premier vu que c'est plus ou moin un hot fix

Reply:
> Peut etre que c'est status label qui n'est pas recréer ! Et vu qu'on set la var dans le controller et que c'est cette var qu'on uttilise en suite ! La modification du status par svelte n'a pas d'impacte sur l'affichage !

Oui c'est ça qu'il se passe, et le `#key` est déjà une solution pour re-créer le composant quand on change la valeur. Et du coup je comprends pas pourquoi tu proposes encore d'autres solutions.

> sois uttiliser en direct la property dans le componant svelte 

Ça casse notre archi de manière pas nécessaire

> sois avoir un store qui ecoute le changement de valeur !

Encore une fois pas nécessaire, si ?

Reply:
En faite je me demande si c'est au parent de gérer ça ou bien si c'est à status label d'être dynamique ?

Reply:
En soit c'est un composant d'affichage, je ne pense pas qu'il soit censé être dynamique

Reply:
J'étais plutôt surprise qu'il ne soit pas recréé

Reply:
Ok si on part du principe que le soucis viens de svelte ça me va ! Tu peut juste mettre un commentaire qui explique que le composans n'est pas recréer

Reply:
@mxmeunier si je fais ce commentaire est-ce que ça résout ta question à toi ?